Tests/RoslynPluginGeneratorTests/NuGetPackageHandlerTests.cs

```diff
@@ -17,22 +19,211 @@ public class NuGetPackageHandlerTests
     {
         public TestContext TestContext { get; set; }
 
+        private const string testPackageName = "testPackage";
+        private const string dependentPackageName = "dependentPackage";
+
+        private const string releaseVersion = "1.0.0";
+        private const string preReleaseVersion = "1.0.0-RC1";
+
         [TestMethod]
-        public void NuGet_TestDependencyResolutionFailure()
+        public void NuGet_TestPackageDownload_Release_Release()
         {
             // Arrange
             string testDir = TestUtils.CreateTestDirectory(this.TestContext);
+            string testDownloadDir = Path.Combine(testDir, "download");
```

Comment:
Looks like a lot of the setup is duplicated between these tests. Consider refactoring into a separate method.